File: rust/examples/basic_implementation/example_search.rs
#[path = "schema.rs"]
//Load the schema file
mod schema;

use skypydb::{json_map, ReactiveClient, Result};

fn main() -> Result<()> {
    //Create a new client
    let client = ReactiveClient::new("./db/_generated/skypydb_rust.db", None, None, None)?;
    let schema = schema::build_schema()?;

    //Get or create tables from the schema
    let tables = client.get_or_create_tables(&schema)?;

    //Error handling for tables
    let success_table = tables
        .get("success")
        .ok_or_else(|| skypydb::SkypydbError::table_not_found("Table 'success' not found"))?;

    //Search for data in the table
    let results = success_table.search(
        None,
        json_map! {
            "user_id" => "user123",
        },
    )?;

    //Print the results
    if results.is_empty() {
        println!("No results found.");
    } else {
        for item in results {
            println!("{item}");
        }
    }

    Ok(())
}

File: demo/examples/python/basic_implementation/example_search.py
import skypydb

# Create a client
client = skypydb.ReactiveClient()

# Create tables from the schema
# This reads the schema from db/schema.py and creates all tables
tables = client.get_or_create_table()
# if the tables already exists the programe get them instead

# Access your tables
success_table = tables["success"]
warning_table = tables["warning"]
error_table = tables["error"]

# Ensure there is at least one row to demonstrate search
success_table.add(
    component="AuthService",
    action="login",
    message="User logged in successfully",
    user_id="user123"
)

# Search results by filter
user_success_logs = success_table.search(
    user_id="user123"
)

if not user_success_logs:
    print("No results found.")
else:
    for user_success_log in user_success_logs:
        print(user_success_log)
